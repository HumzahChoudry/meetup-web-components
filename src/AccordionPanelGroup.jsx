import React from 'react';
import cx from 'classnames';

export const ACCORDIONPANELGROUP_CLASS = 'accordionPanelGroup';
/**
 * @module AccordionPanelGroup
 */
class AccordionPanelGroup extends React.Component {
	constructor(props) {
		super(props);
		this.accordionPanels = this.props.accordionPanels.map((accordionPanel, i) => {
			return this.clonePanels(accordionPanel, i, accordionPanel.props.isOpen);
		});

		this.state = {
			clickedPanel: null
		};

		this.clonePanels = this.clonePanels.bind(this);
		this.setClickedPanel = this.setClickedPanel.bind(this);
	}

	/**
	 * @param clickedPanel is the `AccordionPanel` component passed in from `_handleToggle`
	 * @returns {undefined}
	 *
	 * Keeps track of the clicked panel in order to support AccordionPanelGroups that only
	 * have one panel open at a time.
	 */
	setClickedPanel(clickedPanel) {
		this.setState({ clickedPanel });
	}

	/**
	 * @param {Object} accordionPanel - `AccordionPanel` components to clone
	 * @param {number} i - index of the `AccordionPanel`
	 * @param {boolean} isOpen - whether the `AccordionPanel` is open or not
	 * @returns {Array} `AccordionPanel` components with props from `AccordionPanelGroup`
	 */
	clonePanels(accordionPanel, i, isOpen) {
		const panelProps = {
			key: i,
			panelId: i,
			isAnimated: this.props.isAnimated,
			className: accordionPanel.props.className,
			setClickedPanel: this.props.multiSelectable ? false : this.setClickedPanel,
			isOpen: isOpen
		};

		if (this.props.iconOptions) {
			panelProps['icon'] = this.props.iconOptions;
		}

		return React.cloneElement(accordionPanel, panelProps);
	}

	/**
	 * @returns {Array} `AccordionPanel` components with the correct value for `isOpen` prop
	 */
	renderAccordionPanels() {
		this.accordionPanels = this.accordionPanels.map((accordionPanel, i) => {
			const isOpen = accordionPanel.props.isOpen && this.state.clickedPanel == null;

			return this.clonePanels(accordionPanel, i, isOpen);
		});

		return this.accordionPanels;
	}

	render() {
		const {
			isAnimated, // eslint-disable-line no-unused-vars
			accordionPanels, // eslint-disable-line no-unused-vars
			iconOptions, // eslint-disable-line no-unused-vars
			multiSelectable,
			className,
			...other
		} = this.props;

		const classNames = cx(
			ACCORDIONPANELGROUP_CLASS,
			'list',
			className
		);

		return (
			<ul
				role='tabList'
				aria-multiselectable={multiSelectable}
				className={classNames}
				{...other}
				>
					{this.renderAccordionPanels()}
			</ul>
		);
	}
}

AccordionPanelGroup.defaultProps = {
	multiSelectable: true,
};

AccordionPanelGroup.propTypes = {
	accordionPanels: React.PropTypes.arrayOf(React.PropTypes.element).isRequired,
	multiSelectable: React.PropTypes.bool,
	iconOptions: React.PropTypes.shape({
		align: React.PropTypes.string,
		shape: React.PropTypes.string,
		shapeActive: React.PropTypes.string,
		size: React.PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl']),
	})
};

export default AccordionPanelGroup;
