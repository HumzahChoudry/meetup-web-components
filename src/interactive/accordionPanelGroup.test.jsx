import React from 'react';
import { mount } from 'enzyme';

import AccordionPanelGroup from './AccordionPanelGroup';
import AccordionPanel, { PANEL_CLASS } from './AccordionPanel';

describe('AccordionPanelGroup', () => {

	const accordionPanelsArr = [
		<AccordionPanel
			label='First Section'
			isOpen
			panelContent={
				<div className='runningText'>
					<p>Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.</p>
				</div>
			} />,
		<AccordionPanel
			label='Next Section'
			panelContent={
				<div>
					<div className='runningText'>
						<p>Any kind of content can go in here, even inputs.</p>
					</div>
					<div className='chunk'>
						<label htmlFor='test-textinput'>I'm a label</label>
						<input id='test-textinput' type='text' placeholder='Input placeholder' />
					</div>
				</div>
			} />,
		<AccordionPanel
			label='Third Section'
			panelContent={
				<div className='runningText'>
					<p>Classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.</p>
				</div>
			} />,
	];

	describe('AccordionPanelGroup, multiselectable', () => {
		let accordionPanelGroup;

		beforeEach(() => {
			accordionPanelGroup = mount(
				<AccordionPanelGroup accordionPanels={accordionPanelsArr} />
			);
		});

		afterEach(() => {
			accordionPanelGroup = null;
		});

		it('should render panels from array', function() {
			expect(accordionPanelGroup).toMatchSnapshot();
		});

		it('should render all the panels given', function() {
			const panels = accordionPanelGroup.find(`.${PANEL_CLASS}`);
			expect(panels.length).toEqual(3);
		});

		it('should give panels a clickId', () => {
			expect(accordionPanelGroup).toMatchSnapshot();
			const panels = accordionPanelGroup.find(AccordionPanel);
			expect(panels.at(0).prop('clickId')).not.toBeUndefined();
			expect(panels.at(1).prop('clickId')).not.toBeUndefined();
			expect(panels.at(2).prop('clickId')).not.toBeUndefined();
		});

		it('should store panel open states in state', () => {
			// should have one open panel to start based on isOpen prop
			const isOpenValues = Object.values(accordionPanelGroup.state('panelStates'));
			const openPanels = isOpenValues.filter((isOpen, i) => isOpen === true);
			const closedPanels = isOpenValues.filter((isOpen, i) => isOpen === false);

			expect(openPanels.length).toEqual(1);
			expect(closedPanels.length).toEqual(2);
		});

		it('should change panel open state with setPanelStates', () => {
			const panels = accordionPanelGroup.find(AccordionPanel);
			const clickId = panels.at(0).prop('clickId');
			const isOpen = panels.at(0).prop('isOpen');

			expect(accordionPanelGroup.state('panelStates')[clickId]).toEqual(isOpen);
			accordionPanelGroup.instance().setPanelStates(clickId, !isOpen);
			expect(accordionPanelGroup.state('panelStates')[clickId]).toEqual(!isOpen);
		});

	});

	describe('AccordionPanelGroup, not multiselectable', () => {
		let accordionPanelGroup,
			setPanelStatesMock;

		beforeEach(() => {
			setPanelStatesMock = jest.spyOn(AccordionPanelGroup.prototype, 'setPanelStates');

			accordionPanelGroup = mount(
				<AccordionPanelGroup accordionPanels={accordionPanelsArr} multiSelectable={false} />
			);
		});

		afterEach(() => {
			accordionPanelGroup = null;
			setPanelStatesMock = null;
		});

		it('should support opening panels one-at-a-time', function(){
			const panelWrappers = accordionPanelGroup.find(AccordionPanel);
			let openPanels = panelWrappers.filterWhere((panel) => panel.prop('isOpen') === true);

			// there is one panel in our test with prop isOpen at first render
			expect(openPanels.length).toBe(1);

			const panel1 = panelWrappers.at(1);
			panel1.find('button').simulate('click');
			openPanels = panelWrappers.filterWhere((panel) => panel.prop('isOpen') === true);
			expect(openPanels.length).toBe(1);
			expect(openPanels.at(0).prop('clickId')).toEqual(panel1.prop('clickId'));

			const panel2 = panelWrappers.at(2);
			panel2.find('button').simulate('click');
			openPanels = panelWrappers.filterWhere((panel) => panel.prop('isOpen') === true);
			expect(openPanels.length).toBe(1);
			expect(openPanels.at(0).prop('clickId')).toEqual(panel2.prop('clickId'));
		});

		it('should have state reflecting support opening panels one-at-a-time', () => {
			const panelWrappers = accordionPanelGroup.find(AccordionPanel);
			const openPanels = panelWrappers.filterWhere((panel) => panel.prop('isOpen') === true);
			expect(openPanels.length).toBe(1);

			let panelStates = accordionPanelGroup.state('panelStates');
			let openPanelIds = Object.keys(panelStates).filter((key, i) => (panelStates[key] === true));
			expect(openPanelIds.length).toBe(1);

			expect(parseInt(openPanelIds[0])).toEqual(openPanels.at(0).prop('clickId'));

			// we have one with prop isOpen
			const panel1 = panelWrappers.at(1);
			panel1.find('button').simulate('click');
			expect(setPanelStatesMock).toHaveBeenCalled();

			// assert only one key in panelStates with isOpen and matching clickId
			panelStates = accordionPanelGroup.state('panelStates');
			openPanelIds = Object.keys(panelStates).filter((key, i) => panelStates[key] === true);
			expect(openPanelIds.length).toBe(1);
			expect(parseInt(openPanelIds[0])).toEqual(panel1.prop('clickId'));
		});
	});
});
