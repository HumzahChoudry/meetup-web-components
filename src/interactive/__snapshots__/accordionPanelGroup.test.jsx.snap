// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionPanelGroup AccordionPanelGroup, multiselectable gives panels a clickId 1`] = `
<AccordionPanelGroup
  accordionPanels={
    Array [
      <AccordionPanel
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconSize="xs"
        isOpen={true}
        label="First Section"
        panelContent={
            <div
                className="runningText"
            >
                <p>
                    Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                </p>
            </div>
        }
    />,
      <AccordionPanel
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconSize="xs"
        isOpen={false}
        label="Next Section"
        panelContent={
            <div>
                <div
                    className="runningText"
                >
                    <p>
                        Any kind of content can go in here, even inputs.
                    </p>
                </div>
                <div
                    className="chunk"
                >
                    <label
                        htmlFor="test-textinput"
                    >
                        Im a label
                    </label>
                    <input
                        id="test-textinput"
                        placeholder="Input placeholder"
                        type="text"
                    />
                </div>
            </div>
        }
    />,
      <AccordionPanel
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconSize="xs"
        isOpen={false}
        label="Third Section"
        panelContent={
            <div
                className="runningText"
            >
                <p>
                    Classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                </p>
            </div>
        }
    />,
    ]
  }
  indicatorAlign="right"
  indicatorIcon="chevron-down"
  indicatorIconSize="xs"
  multiSelectable={true}
>
  <ul
    aria-multiselectable={true}
    className="accordionPanelGroup list"
    role="tabList"
  >
    <li
      className="list-item flush--top"
      key="0"
    >
      <AccordionPanel
        clickId={0}
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconActive={undefined}
        indicatorIconSize="xs"
        indicatorSwitch={undefined}
        isOpen={true}
        label="First Section"
        panelContent={
          <div
            className="runningText"
          >
            <p>
              Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
            </p>
          </div>
        }
        setClickedPanel={[Function]}
      >
        <div>
          <button
            aria-controls="panel-firstsection"
            aria-expanded={true}
            aria-selected={true}
            className="accordionPanel-label display--block span--100"
            id="label-firstsection"
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            role="tab"
          >
            <Flex
              className="padding--bottom padding--top accordionPanel accordionPanel--active"
              direction="row"
              rowReverse={false}
              tabIndex={-1}
            >
              <div
                className="flex flex--row padding--bottom padding--top accordionPanel accordionPanel--active"
                tabIndex={-1}
              >
                <FlexItem>
                  <div
                    className="flex-item"
                  >
                    First Section
                  </div>
                </FlexItem>
                <FlexItem
                  className="accordionPanel-icon"
                  onClick={[Function]}
                  shrink={true}
                >
                  <div
                    className="flex-item flex-item--shrink accordionPanel-icon"
                    onClick={[Function]}
                  >
                    <Icon
                      shape="chevron-down"
                      size="xs"
                    >
                      <span
                        className="svg svg--chevron-down"
                      >
                        <svg
                          className="svg-icon valign--middle"
                          height="16"
                          preserveAspectRatio="xMinYMin meet"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <use
                            xlinkHref="#icon-chevron-down--small"
                          />
                        </svg>
                      </span>
                    </Icon>
                  </div>
                </FlexItem>
              </div>
            </Flex>
          </button>
          <Chunk
            aria-hidden={false}
            aria-labelledby="label-firstsection"
            className="accordionPanel-animator"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "auto",
              }
            }
          >
            <div
              aria-hidden={false}
              aria-labelledby="label-firstsection"
              className="chunk accordionPanel-animator"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "auto",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div
                  className="runningText"
                >
                  <p>
                    Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                  </p>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </AccordionPanel>
    </li>
    <li
      className="list-item flush--top"
      key="1"
    >
      <AccordionPanel
        clickId={1}
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconActive={undefined}
        indicatorIconSize="xs"
        indicatorSwitch={undefined}
        isOpen={false}
        label="Next Section"
        panelContent={
          <div>
            <div
              className="runningText"
            >
              <p>
                Any kind of content can go in here, even inputs.
              </p>
            </div>
            <div
              className="chunk"
            >
              <label
                htmlFor="test-textinput"
              >
                Im a label
              </label>
              <input
                id="test-textinput"
                placeholder="Input placeholder"
                type="text"
              />
            </div>
          </div>
        }
        setClickedPanel={[Function]}
      >
        <div>
          <button
            aria-controls="panel-nextsection"
            aria-expanded={false}
            aria-selected={false}
            className="accordionPanel-label display--block span--100"
            id="label-nextsection"
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            role="tab"
          >
            <Flex
              className="padding--bottom padding--top accordionPanel"
              direction="row"
              rowReverse={false}
              tabIndex={-1}
            >
              <div
                className="flex flex--row padding--bottom padding--top accordionPanel"
                tabIndex={-1}
              >
                <FlexItem>
                  <div
                    className="flex-item"
                  >
                    Next Section
                  </div>
                </FlexItem>
                <FlexItem
                  className="accordionPanel-icon"
                  onClick={[Function]}
                  shrink={true}
                >
                  <div
                    className="flex-item flex-item--shrink accordionPanel-icon"
                    onClick={[Function]}
                  >
                    <Icon
                      shape="chevron-down"
                      size="xs"
                    >
                      <span
                        className="svg svg--chevron-down"
                      >
                        <svg
                          className="svg-icon valign--middle"
                          height="16"
                          preserveAspectRatio="xMinYMin meet"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <use
                            xlinkHref="#icon-chevron-down--small"
                          />
                        </svg>
                      </span>
                    </Icon>
                  </div>
                </FlexItem>
              </div>
            </Flex>
          </button>
          <Chunk
            aria-hidden={true}
            aria-labelledby="label-nextsection"
            className="accordionPanel-animator accordionPanel-animator--collapse"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "0px",
              }
            }
          >
            <div
              aria-hidden={true}
              aria-labelledby="label-nextsection"
              className="chunk accordionPanel-animator accordionPanel-animator--collapse"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "0px",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div>
                  <div
                    className="runningText"
                  >
                    <p>
                      Any kind of content can go in here, even inputs.
                    </p>
                  </div>
                  <div
                    className="chunk"
                  >
                    <label
                      htmlFor="test-textinput"
                    >
                      Im a label
                    </label>
                    <input
                      id="test-textinput"
                      placeholder="Input placeholder"
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </AccordionPanel>
    </li>
    <li
      className="list-item flush--top"
      key="2"
    >
      <AccordionPanel
        clickId={2}
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconActive={undefined}
        indicatorIconSize="xs"
        indicatorSwitch={undefined}
        isOpen={false}
        label="Third Section"
        panelContent={
          <div
            className="runningText"
          >
            <p>
              Classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
            </p>
          </div>
        }
        setClickedPanel={[Function]}
      >
        <div>
          <button
            aria-controls="panel-thirdsection"
            aria-expanded={false}
            aria-selected={false}
            className="accordionPanel-label display--block span--100"
            id="label-thirdsection"
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            role="tab"
          >
            <Flex
              className="padding--bottom padding--top accordionPanel"
              direction="row"
              rowReverse={false}
              tabIndex={-1}
            >
              <div
                className="flex flex--row padding--bottom padding--top accordionPanel"
                tabIndex={-1}
              >
                <FlexItem>
                  <div
                    className="flex-item"
                  >
                    Third Section
                  </div>
                </FlexItem>
                <FlexItem
                  className="accordionPanel-icon"
                  onClick={[Function]}
                  shrink={true}
                >
                  <div
                    className="flex-item flex-item--shrink accordionPanel-icon"
                    onClick={[Function]}
                  >
                    <Icon
                      shape="chevron-down"
                      size="xs"
                    >
                      <span
                        className="svg svg--chevron-down"
                      >
                        <svg
                          className="svg-icon valign--middle"
                          height="16"
                          preserveAspectRatio="xMinYMin meet"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <use
                            xlinkHref="#icon-chevron-down--small"
                          />
                        </svg>
                      </span>
                    </Icon>
                  </div>
                </FlexItem>
              </div>
            </Flex>
          </button>
          <Chunk
            aria-hidden={true}
            aria-labelledby="label-thirdsection"
            className="accordionPanel-animator accordionPanel-animator--collapse"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "0px",
              }
            }
          >
            <div
              aria-hidden={true}
              aria-labelledby="label-thirdsection"
              className="chunk accordionPanel-animator accordionPanel-animator--collapse"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "0px",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div
                  className="runningText"
                >
                  <p>
                    Classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                  </p>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </AccordionPanel>
    </li>
  </ul>
</AccordionPanelGroup>
`;

exports[`AccordionPanelGroup AccordionPanelGroup, multiselectable renders panels from array 1`] = `
<AccordionPanelGroup
  accordionPanels={
    Array [
      <AccordionPanel
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconSize="xs"
        isOpen={true}
        label="First Section"
        panelContent={
            <div
                className="runningText"
            >
                <p>
                    Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                </p>
            </div>
        }
    />,
      <AccordionPanel
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconSize="xs"
        isOpen={false}
        label="Next Section"
        panelContent={
            <div>
                <div
                    className="runningText"
                >
                    <p>
                        Any kind of content can go in here, even inputs.
                    </p>
                </div>
                <div
                    className="chunk"
                >
                    <label
                        htmlFor="test-textinput"
                    >
                        Im a label
                    </label>
                    <input
                        id="test-textinput"
                        placeholder="Input placeholder"
                        type="text"
                    />
                </div>
            </div>
        }
    />,
      <AccordionPanel
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconSize="xs"
        isOpen={false}
        label="Third Section"
        panelContent={
            <div
                className="runningText"
            >
                <p>
                    Classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                </p>
            </div>
        }
    />,
    ]
  }
  indicatorAlign="right"
  indicatorIcon="chevron-down"
  indicatorIconSize="xs"
  multiSelectable={true}
>
  <ul
    aria-multiselectable={true}
    className="accordionPanelGroup list"
    role="tabList"
  >
    <li
      className="list-item flush--top"
      key="0"
    >
      <AccordionPanel
        clickId={0}
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconActive={undefined}
        indicatorIconSize="xs"
        indicatorSwitch={undefined}
        isOpen={true}
        label="First Section"
        panelContent={
          <div
            className="runningText"
          >
            <p>
              Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
            </p>
          </div>
        }
        setClickedPanel={[Function]}
      >
        <div>
          <button
            aria-controls="panel-firstsection"
            aria-expanded={true}
            aria-selected={true}
            className="accordionPanel-label display--block span--100"
            id="label-firstsection"
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            role="tab"
          >
            <Flex
              className="padding--bottom padding--top accordionPanel accordionPanel--active"
              direction="row"
              rowReverse={false}
              tabIndex={-1}
            >
              <div
                className="flex flex--row padding--bottom padding--top accordionPanel accordionPanel--active"
                tabIndex={-1}
              >
                <FlexItem>
                  <div
                    className="flex-item"
                  >
                    First Section
                  </div>
                </FlexItem>
                <FlexItem
                  className="accordionPanel-icon"
                  onClick={[Function]}
                  shrink={true}
                >
                  <div
                    className="flex-item flex-item--shrink accordionPanel-icon"
                    onClick={[Function]}
                  >
                    <Icon
                      shape="chevron-down"
                      size="xs"
                    >
                      <span
                        className="svg svg--chevron-down"
                      >
                        <svg
                          className="svg-icon valign--middle"
                          height="16"
                          preserveAspectRatio="xMinYMin meet"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <use
                            xlinkHref="#icon-chevron-down--small"
                          />
                        </svg>
                      </span>
                    </Icon>
                  </div>
                </FlexItem>
              </div>
            </Flex>
          </button>
          <Chunk
            aria-hidden={false}
            aria-labelledby="label-firstsection"
            className="accordionPanel-animator"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "auto",
              }
            }
          >
            <div
              aria-hidden={false}
              aria-labelledby="label-firstsection"
              className="chunk accordionPanel-animator"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "auto",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div
                  className="runningText"
                >
                  <p>
                    Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                  </p>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </AccordionPanel>
    </li>
    <li
      className="list-item flush--top"
      key="1"
    >
      <AccordionPanel
        clickId={1}
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconActive={undefined}
        indicatorIconSize="xs"
        indicatorSwitch={undefined}
        isOpen={false}
        label="Next Section"
        panelContent={
          <div>
            <div
              className="runningText"
            >
              <p>
                Any kind of content can go in here, even inputs.
              </p>
            </div>
            <div
              className="chunk"
            >
              <label
                htmlFor="test-textinput"
              >
                Im a label
              </label>
              <input
                id="test-textinput"
                placeholder="Input placeholder"
                type="text"
              />
            </div>
          </div>
        }
        setClickedPanel={[Function]}
      >
        <div>
          <button
            aria-controls="panel-nextsection"
            aria-expanded={false}
            aria-selected={false}
            className="accordionPanel-label display--block span--100"
            id="label-nextsection"
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            role="tab"
          >
            <Flex
              className="padding--bottom padding--top accordionPanel"
              direction="row"
              rowReverse={false}
              tabIndex={-1}
            >
              <div
                className="flex flex--row padding--bottom padding--top accordionPanel"
                tabIndex={-1}
              >
                <FlexItem>
                  <div
                    className="flex-item"
                  >
                    Next Section
                  </div>
                </FlexItem>
                <FlexItem
                  className="accordionPanel-icon"
                  onClick={[Function]}
                  shrink={true}
                >
                  <div
                    className="flex-item flex-item--shrink accordionPanel-icon"
                    onClick={[Function]}
                  >
                    <Icon
                      shape="chevron-down"
                      size="xs"
                    >
                      <span
                        className="svg svg--chevron-down"
                      >
                        <svg
                          className="svg-icon valign--middle"
                          height="16"
                          preserveAspectRatio="xMinYMin meet"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <use
                            xlinkHref="#icon-chevron-down--small"
                          />
                        </svg>
                      </span>
                    </Icon>
                  </div>
                </FlexItem>
              </div>
            </Flex>
          </button>
          <Chunk
            aria-hidden={true}
            aria-labelledby="label-nextsection"
            className="accordionPanel-animator accordionPanel-animator--collapse"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "0px",
              }
            }
          >
            <div
              aria-hidden={true}
              aria-labelledby="label-nextsection"
              className="chunk accordionPanel-animator accordionPanel-animator--collapse"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "0px",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div>
                  <div
                    className="runningText"
                  >
                    <p>
                      Any kind of content can go in here, even inputs.
                    </p>
                  </div>
                  <div
                    className="chunk"
                  >
                    <label
                      htmlFor="test-textinput"
                    >
                      Im a label
                    </label>
                    <input
                      id="test-textinput"
                      placeholder="Input placeholder"
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </AccordionPanel>
    </li>
    <li
      className="list-item flush--top"
      key="2"
    >
      <AccordionPanel
        clickId={2}
        indicatorAlign="right"
        indicatorIcon="chevron-down"
        indicatorIconActive={undefined}
        indicatorIconSize="xs"
        indicatorSwitch={undefined}
        isOpen={false}
        label="Third Section"
        panelContent={
          <div
            className="runningText"
          >
            <p>
              Classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
            </p>
          </div>
        }
        setClickedPanel={[Function]}
      >
        <div>
          <button
            aria-controls="panel-thirdsection"
            aria-expanded={false}
            aria-selected={false}
            className="accordionPanel-label display--block span--100"
            id="label-thirdsection"
            onBlur={undefined}
            onClick={[Function]}
            onFocus={undefined}
            role="tab"
          >
            <Flex
              className="padding--bottom padding--top accordionPanel"
              direction="row"
              rowReverse={false}
              tabIndex={-1}
            >
              <div
                className="flex flex--row padding--bottom padding--top accordionPanel"
                tabIndex={-1}
              >
                <FlexItem>
                  <div
                    className="flex-item"
                  >
                    Third Section
                  </div>
                </FlexItem>
                <FlexItem
                  className="accordionPanel-icon"
                  onClick={[Function]}
                  shrink={true}
                >
                  <div
                    className="flex-item flex-item--shrink accordionPanel-icon"
                    onClick={[Function]}
                  >
                    <Icon
                      shape="chevron-down"
                      size="xs"
                    >
                      <span
                        className="svg svg--chevron-down"
                      >
                        <svg
                          className="svg-icon valign--middle"
                          height="16"
                          preserveAspectRatio="xMinYMin meet"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <use
                            xlinkHref="#icon-chevron-down--small"
                          />
                        </svg>
                      </span>
                    </Icon>
                  </div>
                </FlexItem>
              </div>
            </Flex>
          </button>
          <Chunk
            aria-hidden={true}
            aria-labelledby="label-thirdsection"
            className="accordionPanel-animator accordionPanel-animator--collapse"
            onTransitionEnd={[Function]}
            role="tabpanel"
            style={
              Object {
                "height": "0px",
              }
            }
          >
            <div
              aria-hidden={true}
              aria-labelledby="label-thirdsection"
              className="chunk accordionPanel-animator accordionPanel-animator--collapse"
              onTransitionEnd={[Function]}
              role="tabpanel"
              style={
                Object {
                  "height": "0px",
                }
              }
            >
              <div
                className="accordionPanel-content"
              >
                <div
                  className="runningText"
                >
                  <p>
                    Classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.
                  </p>
                </div>
              </div>
            </div>
          </Chunk>
        </div>
      </AccordionPanel>
    </li>
  </ul>
</AccordionPanelGroup>
`;
